src/Integration.Vsix.UnitTests/SonarLintTagger/TextBufferIssueTrackerTests.cs

```diff
@@ -243,6 +244,36 @@ public void WhenNewIssuesAreFound_ListenersAreUpdated()
             actualMarkers[1].Issue.RuleKey.Should().Be("S333");
         }
 
+        [TestMethod]
+        public void WhenNewIssuesAreFound_IssueFilterIsApplied()
+        {
+            // Arrange
+            var inputIssues = new[] {
+                new Sonarlint.Issue { RuleKey = "S111", StartLine = 1, EndLine = 1 },
+                new Sonarlint.Issue { RuleKey = "S222", StartLine = 2, EndLine = 2 },
+                new Sonarlint.Issue { RuleKey = "S333", StartLine = 10, EndLine = 20 },
+                new Sonarlint.Issue { RuleKey = "S444", StartLine = 100, EndLine = 101 }
+            };
+
+            // Set up the filter to return only one issue
+            var originalIssues = new List<IFilterableIssue>();
+            issuesFilter.Setup(x => x.Filter(It.IsAny<string>(), It.IsAny<IEnumerable<IFilterableIssue>>()))
+                .Callback((string path, IEnumerable<IFilterableIssue> issues) => originalIssues.AddRange(issues))
+                .Returns(originalIssues.Where(i => i.RuleId == "S222" || i.RuleId == "S444"));
```

Comment:
I think there are more conditions that we should check here, e.g.:
0/1/2 issues returned, null issue check, suppressions log, etc.
I think this will be much easier to test once it's in a separate class.